BSB_lan.ino

```diff
@@ -8046,6 +8053,124 @@ uint8_t pps_offset = 0;
     }
   }
 } // --- loop () ---
+//Luposoft: Funktionen mqtt_connect
+/*  Function: mqtt_connect()
+ *  Does:     connect to mqtt broker
+ 
+ * Pass parameters:
+ *  none
+ * Parameters passed back:
+ *  none
+ * Function value returned:
+ *  boolean
+ * Global resources used:
+ *  Serial instance
+ *  Ethernet instance
+ *  MQTT instance
+ * *************************************************************** */
+#ifdef MQTT
+boolean mqtt_connect()
+{
+  if (!MQTTClient->connected())
+  {
+    IPAddress MQTTBroker(mqtt_broker_ip_addr[0], mqtt_broker_ip_addr[1], mqtt_broker_ip_addr[2], mqtt_broker_ip_addr[3]);
+    MQTTClient->setServer(MQTTBroker, 1883);
+    int retries = 0;
+    while (!MQTTClient->connected() && retries < 3)
+    {
+      MQTTClient->connect("BSB-LAN", MQTTUser, MQTTPass);
+      retries++;
+      if (!MQTTClient->connected())
+      {
+        delay(1000);
+        printlnToDebug(PSTR("Failed to connect to MQTT broker, retrying..."));
+      } 
+      else
+      {
+        printlnToDebug(PSTR("Connect to MQTT broker"));
+        MQTTClient->subscribe("fromFHEM");   //Luposoft: set the topic listen to
```

Comment:
Need to add new variable or use existing MQTTTopicPrefix instead static-defined "fromFHEM"

Reply:
I don't think so, but why not?
Now I use the MQTTTopicPrefix for standard, if empty 'FromFhem'

Reply:
Main software is FHEM but here can be other MQTT client/servers.

Reply:
You are right. But there must be a standard value when not defined MQTTTopicPrefix.
What do you think about 'fromBroker' ?

Reply:
I think we need to ask @fredlcore

Reply:
Up to me, "fromBroker" is definitely much better than "fromFHEM"! The meaning of "broker" is 100% clear but "FHEM" isn't for someone who doesn't know it because of the usage of a different home automation system. So I would say: Please go with "fromBroker"..